game-app/game-core/src/main/java/games/strategy/triplea/ui/UnitChooser.java

```diff
@@ -25,11 +34,24 @@
 import javax.swing.JLabel;
 import javax.swing.JPanel;
 import javax.swing.JTextArea;
+import org.triplea.java.Postconditions;
 import org.triplea.java.collections.IntegerMap;
 
+/**
+ * This is the <code>JPanel</code> that shows the units to choose
+ * including the controls to make the choice.
+ *
+ * It's used by many different use cases like placing units or
+ * choosing which units take hits in a battle round.
+ */
+
 public final class UnitChooser extends JPanel {
   private static final long serialVersionUID = -4667032237550267682L;
-  private final List<ChooserEntry> entries = new ArrayList<>();
+
+  private static final Insets nullInsets = new Insets(0, 0, 0, 0);
+
+  @VisibleForTesting
+  public final List<ChooserEntry> entries = new ArrayList<>();
```

Comment:
It's notable to see a public variable that is visible for testing.

It does not look like this variable needs to be 'public', could you se this back to private?

Reply:
I need it in my ui tests. Can we include this issue in the ui testing discussion which we agreed to postpone?

Reply:
It would be nice to see those tests land cleanly and then see which changes are then needed for it to work. That way we can discuss cleaner ways to do this. Direct access to a variable is a violation of encapsulation, it's a pretty big code smell and I'd be concerned the test would be overly coupled.

With that said, we need to get this done. If it's a big burden to undo this update and then redo it, then please do not worry about it.